.gitlab-ci.yml

```diff
@@ -581,20 +586,30 @@ validate:edge+flambda:
 # but not on its ML API
 
 library:ci-argosy:
-  extends: .ci-template
+  extends: .ci-template-flambda
```

Comment:
I am not sure whether it is a good idea to make the whole CI test flambda only. It is hardly used by our users (?) and it has been removed from recent OCaml compilers. The only good argument I can imagine for it is a noticeable decrease of CI resources, but is that even the case?

Reply:
Even if it's not exactly the whole CI (there are a few things that don't depend on stdlib, like HoTT, Unimath, some plugins,...) it's indeed a large part of it.
My main reason for the change was that stdlib requires dune 3.7 and the base image has only dune 3.6.1 but I guess I could also just bump the dune version there.

Reply:
Not so obvious to me. The base image should ideally test the minimal supported version for things in this repository, and this remains dune 3.6.1. But maybe it would make sense to bump to dune 3.7.
I'll let our dune guru decide @ejgallego @SkySkimmer , would it be ok to bump the minimal dune required version from 3.6.1 to 3.7 ?

Reply:
I don't see any problem doing that, in fact the version we recommend these days is 3.8 , which supports the 0.8 version of the language that will, with minor changes, become 1.0 unless we detect a critical problem.

I was waiting to bump to 3.8 until later, as to give a bit more of buffer time.

Reply:
Yes, I was indeed implicitly suggesting a bump of the minimal dune version to match what will be needed for the stdlib (no need to artificially keep the two at different version numbers, given that virtually all users will need to depend on the stdlib).

Some other arguments in favor of bumping:

- Dune 3.7 was released early 2023, so ~2 years before the next Coq release.
- Debian testing includes Dune 3.16 and Debian stable only includes Dune 2.9.3 (so anyway too old for us).
- Similarly, the first version of Ubuntu with Dune 3 is Ubuntu 23.10 and it includes Dune 3.10.

Reply:
Here it is https://github.com/coq/coq/pull/19621